startup.d/10-git-pull.sh

```diff
@@ -1,2 +1,21 @@
 #!/bin/sh
+
+# Prepare code auditing environment
+REPO_URL=$(git remote get-url origin)
+TEST_DIR=$(mktemp -d)
+git clone "$REPO_URL" "$TEST_DIR";
+pushd "$TEST_DIR";
+
+# Code auditing section
+python -m py_compile $(find . -name '*.py');
```

Comment:
I mentioned flake8 because it's what I use. I can see why you'd not want to fail on style though. Maybe use pyflakes directly instead? Regardless, like I said above, this is still better than what we have. We can refine it later.

You really do need to use the correct version of python though.